imports/shared/action.models.js

```diff
@@ -0,0 +1,32 @@
+// @flow
+
+import type { RectangularRegion } from '../region/models';
+import type { State } from './state.models';
+
+export type RegionAction = {
```

Comment:
there may be 2 advantages to define a type of an action-payload. 
1. let people understand the data-structure we define. Better for maintenance. 
2. type an action-payload for flow check when we really use redux synchronized action. (`dispatch({type, payload})`).
However, no.2 advantange does not exist since we do not really directly invoke a redux synchronized action. like below, 

```
// https://flow.org/en/docs/react/redux/#toc-typing-redux-thunk-actions
function foo(): ThunkAction {
  return (dispatch, getState) => {
    const baz = getState().baz
    dispatch({ type: "BAR", bar: true }) // we never do this thing
```
Instead, we use mongoUpsert. So the type check for ThunkAction's dispatch' action is useless. 

Regarding no.1, since there are **3 duplicate type/defination** for "regionArray: RectangularRegion[]", 
1. action.models.js: RegionAction
2. state.models.js:RegionState
3. Region.js: RegionComponentProps
so I would suggest to remove the type for RegionAction and also for other actions.